mlflow/transformers/__init__.py

```diff
@@ -2761,6 +2802,32 @@ def is_uri(s):
                 error_code=BAD_REQUEST,
             )
 
+    def _wrap_input_in_prompt_template(self, input_data):
+        """
+        Wraps the input data in the specified prompt template. If no template is
+        specified, or if the pipeline is an unsupported type, or if the input type
+        is not a string or list of strings, then the input data is returned unchanged.
+        """
+
+        # if no template is set, do nothing
+        if not self.prompt_template:
+            return input_data
+
+        if self.pipeline.task not in _SUPPORTED_PROMPT_TEMPLATING_TASK_TYPES:
+            return input_data
+
+        if isinstance(input_data, str):
+            return self.prompt_template.format(prompt=input_data)
+        elif isinstance(input_data, list):
+            # if every item is a string, then apply formatting to every item
+            if all(isinstance(data, str) for data in input_data):
+                return [self.prompt_template.format(prompt=data) for data in input_data]
+            # otherwise, the input might be more complex (e.g. a dict), so we leave it alone
+            else:
+                return input_data
+        else:
+            return input_data
```

Comment:
What's the case where input_data is not str/List[str] but prompt_template is not None? Should we raise a warning?